### PR TITLE
Allow doctest to be invoked via `cabal repl --with-ghc=doctest`

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: test/RunSpec.hs [hunk 1]
```diff
@@ -29,6 +29,14 @@ withCurrentDirectory workingDir action = do
 main :: IO ()
 main = hspec spec
 
+
+removeLoadedPackageEnvironment :: String -> String
+#if __GLASGOW_HASKELL__ < 810
```
Comment (andreasabel) [inline test/RunSpec.hs:34]: Would it hurt to apply this filter unconditionally?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -112,13 +112,14 @@ jobs:
...
           cd "$GITHUB_WORKSPACE"
-          cabal v1-install --enable-tests --only-dep $CABAL_OPTS
 
       - name: script
         run: |
           cd "$GITHUB_WORKSPACE"
-          (cd ghci-wrapper && cabal v1-configure --enable-tests --ghc-options=-Werror $CABAL_OPTS && cabal v1-build && cabal v1-test)
-          cabal v1-configure --enable-tests --ghc-options=-Werror $CABAL_OPTS && cabal v1-build && cabal v1-test
+          (cd ghci-wrapper && cabal configure --enable-tests && cabal build && cabal test --test-show-details=direct)
+          cabal configure --enable-tests && cabal build && cabal exec -- cabal test --test-show-details=direct
```
Comment (andreasabel) [inline .github/workflows/build.yml:120]: If `cabal test` needs to be under `cabal exec`, will `stack` be able to run the tests?

File: test/RunSpec.hs (hunk 1 above)
Comment (sol) [reply test/RunSpec.hs:34]: My primary intention is here is to document why we need this.  So if at some point in the future we drop support for GHC < 8.10 we can remove this code.

File: .github/workflows/build.yml (hunk 2 above)
Comment (sol) [reply .github/workflows/build.yml:120]: I guess it will work with `stack exec` as well.  `doctest` is not really different from `ghci` in that regard.

**Pull request merged**